File: core/src/ru/myitschool/dcrawler/effects/ArcaneShieldingEffect.java
package ru.myitschool.dcrawler.effects;

import com.badlogic.gdx.graphics.Texture;
import ru.myitschool.dcrawler.entities.Entity;

/**
 * Created by Voyager on 26.11.2017.
 */
public class ArcaneShieldingEffect extends Effect {

    private String name = "God's protection";
    private String description = "";
    private Texture icon = new Texture("gods_protection.png");
    private boolean positive = true;

    public ArcaneShieldingEffect(Entity entity) {
        super(entity);
    }

    @Override
    public Texture getIcon() {
        return icon;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDescription() {
        return "God protects you from all damage for this turn";
    }

    @Override
    public boolean isSkillUse() {
        return false;
    }

    @Override
    public boolean isPositive() {
        return true;
    }

    @Override
    public boolean isStackable() {
        return false;
    }

    @Override
    public int getStackSize() {
        return 0;
    }

    @Override
    public boolean isExpiring() {
        return true;
    }

    @Override
    public int getExpireTurns() {
        return 1;
    }

    @Override
    public String getType() {
        return "main.dcrawler.effect.arcane_shielding";
    }

    @Override
    public int onDamage(int damage) {
        return 0;
    }
}

File: core/src/com/wolg_vlad/dcrawler/effects/ArcaneShieldingEffect.kt
package com.wolg_vlad.dcrawler.effects

import com.badlogic.gdx.graphics.Texture
import com.wolg_vlad.dcrawler.entities.Entity

/**
 * Created by Voyager on 26.11.2017.
 */
class ArcaneShieldingEffect(entity: Entity?) : Effect(entity) {
    override val name = "God's protection"
    override val description = "God protects you from all damage for this turn"
    override val icon = Texture("gods_protection.png")
    override val positive: Boolean = true
    override val stackable: Boolean = false
    override val skillUse: Boolean = false
    override val stackSize: Int = 0
    override val expiring: Boolean = true
    override val id: String = "main.dcrawler.effect.arcane_shielding"
    override val expireTurns = 1

    override fun onDamage(damage: Int): Int {
        return 0
    }
}